presentation-api/controlling-ua/PresentationRequest_onconnectionavailable-manual.https.html

```diff
@@ -17,43 +17,57 @@
     // ----------
-    var presentBtn = document.getElementById("presentBtn");
+    const presentBtn = document.getElementById('presentBtn');
 
     // --------------------------------------------------------------------------
     // Start New PresentationRequest.onconnectionavailable Test (success) - begin
     // --------------------------------------------------------------------------
-    var startPresentation = function () {
+    presentBtn.onclick = () => {
         presentBtn.disabled = true;
-        promise_test(function (t) {
-            var connection;
-            t.add_cleanup(function() {
-                if(connection)
-                    connection.terminate();
+
+        promise_test(t => {
+            let connection;
+            const request = new PresentationRequest(presentationUrls);
+
+            t.add_cleanup(() => {
+                if (connection) {
+                    connection.onconnect = () => { connection.terminate(); };
+                    if (connection.state === 'closed')
+                        request.reconnect(connection.id);
+                    else
+                        connection.terminate();
+                }
             });
 
-            // Note: During starting a presentation, the connectionavailable event is fired (step 20)
-            //       after the promise P is resolved (step 19).
-            return new Promise(function(resolve, reject) {
-                var request = new PresentationRequest(presentationUrls);
-                request.onconnectionavailable = function (evt) {
-                    resolve(evt.connection);
-                };
-                // This test fails if request.onconnectionavailable is not invoked although the presentation is started successfully
-                // or the presentation fails to be started
-                request.start().then(function(c) {
-                    connection = c;
-                    t.step_timeout(function() { assert_unreached('The connectionavailable event was not fired.'); }, 5000);
-                }, reject);
-            }).then(function(c) {
+            // Note: During starting a presentation, the connectionavailable event is fired (step 9)
+            //       after the promise P is resolved (step 8).
+            return request.start().then(c => {
                 connection = c;
                 assert_equals(connection.state, 'connecting', 'The initial state of the presentation connection is "connecting".');
                 assert_true(!!connection.id, 'The connection ID is set.');
                 assert_true(typeof connection.id === 'string', 'The connection ID is a string.');
                 assert_true(connection instanceof PresentationConnection, 'The connection is an instance of PresentationConnection.');
+
+                const eventWatcher = new EventWatcher(t, request, 'connectionavailable');
+                const timeout = new Promise(resolve => { t.step_timeout(resolve, 5000); });
```

Comment:
Minor suggestion to improve readability (well, for some definition of readability perhaps): for me, a timeout is something that makes the test fail, so I would translate that into a Promise that gets rejected. In other words, I would write something like:

```js
const timeout = new Promise((_, reject) => { t.step_timeout(() => {
  reject('The connectionavailable event was not fired (timeout)'
}, 5000); });
return Promise.race([ eventWatcher.wait_for('connectionavailable'), timeout ]); 
```

... and then no need to have the `if (!evt)` check in the `then` part.